pwiz_tools/Shared/CommonMsData/RemoteApi/WatersConnect/WatersConnectAccount.cs

```diff
@@ -15,24 +15,71 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
 using System.Net.Http;
+using System.Security.Authentication;
 using System.Xml;
 using System.Xml.Linq;
 using System.Xml.Serialization;
 using IdentityModel.Client;
 using Microsoft.Extensions.DependencyInjection;
+using pwiz.Common;
 using pwiz.Common.SystemUtil;
 
 namespace pwiz.CommonMsData.RemoteApi.WatersConnect
 {
     [XmlRoot("waters_connect_account")]
     public class WatersConnectAccount : RemoteAccount
     {
+        public static readonly string HANDLER_NAME = @"WatersConnect.Handler.Main";
+        public static readonly string AUTH_HANDLER_NAME = @"WatersConnect.Handler.Authentication";
+        public static readonly string TOKEN_DATA = @"token";
+        public static readonly string GET_FOLDERS = @"/waters_connect/v2.0/folders";
+
+        public class TokenCacheEntry
```

Comment:
I wonder if the logic for setting "ExpirationDateTime" should be moved into the constructor since all users of this class calculate it the same way:
```
        public class TokenCacheEntry
        {
            public TokenCacheEntry(TokenResponse tokenResponse)
            {
                TokenResponse = tokenResponse;
                ExpirationDateTime = DateTime.UtcNow.AddSeconds(tokenResponse.ExpiresIn);
            }

            public TokenResponse TokenResponse { get; }
            public DateTime ExpirationDateTime { get; }
        }
```
This would enable you to make the "TokenResponse" and "ExpirationDateTime" properties read-only.